substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -1036,12 +1036,9 @@ impl<'a, E: Ext, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
 		salt_ptr: u32,
 	) -> Result<ReturnErrorCode, TrapReason> {
 		self.charge_gas(RuntimeCosts::Instantiate { input_data_len })?;
-		let deposit_limit: BalanceOf<<E as Ext>::T> = if deposit_ptr == SENTINEL {
-			BalanceOf::<<E as Ext>::T>::zero()
-		} else {
-			memory.read_as(deposit_ptr)?
-		};
-		let value: BalanceOf<<E as Ext>::T> = memory.read_as(value_ptr)?;
+		let deposit_limit: U256 =
+			if deposit_ptr == SENTINEL { U256::zero() } else { memory.read_as(deposit_ptr)? };
+		let value: U256 = memory.read_as(value_ptr)?;
```

Comment:
I really want to get rid of SCALE where possible. Can you replace those instances by just:
```
let mut value = U256::zero();
memory.read_into_buf(value.as_mut_bytes())
```

Saves the memory allocation and doesn't introduce more SCALE dependencies where not necessary. There are other occurences of this.

Reply:
You have to do it a bit differently here, since U256 is defined as a [u64; 4].
Added some default implementation in the Memory trait to read common types (mainly u256, and H256)